testsuite/features/step_definitions/command_steps.rb

```diff
@@ -117,6 +117,49 @@
 end
 
 # Channels
+When(/^I prepare a channel clone for strict mode testing$/) do
+  get_target('server').run('cp -r /srv/www/htdocs/pub/TestRepoRpmUpdates /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST')
+  get_target('server').run('rm -rf /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/repodata')
+  %w[i586 src x86_64].each do |folder|
+    get_target('server').run("rm -f /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/#{folder}/rute-dummy-2.0-1.2.*.rpm")
+  end
+  get_target('server').run('createrepo_c /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST')
+  get_target('server').run('gzip -dc /srv/www/htdocs/pub/TestRepoRpmUpdates/repodata/*-updateinfo.xml.gz > /tmp/updateinfo.xml')
+  get_target('server').run('modifyrepo_c --verbose --mdtype updateinfo /tmp/updateinfo.xml /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/repodata')
+end
+
+Given(/^I am logged into the API$/) do
+  server_node = get_target('server')
+  api_url = "https://#{server_node.public_ip}/rhn/manager/api/auth/login"
+  response = get_target('server').run("curl -H 'Content-Type: application/json' -d '{'login': 'admin', 'password': 'admin'}' -i #{api_url}")
+  raise 'Failed to login to the API' unless response.code == 200
+end
+
+$package_amount = nil
+
+When(/^I check the amount of packages in channel "([^"]*)"$/) do |channel_label|
+  channels = $api_test.channel.list_all_channels
+  if channels.key?(channel_label)
+    package_amount = channels[channel_label]['packages']
+    puts "Package amount for 'test-strict': #{package_amount}"
+  else
+    puts "#{channel_label} channel not found."
+  end
+end
+
+Then(/^The amount of packages in channel "([^"]*)" should be the same as before$/) do |channel_label|
+  channels = $api_test.channel.list_all_channels
+  if channels.key?(channel_label) && ($package_amount != channels[channel_label]['packages'])
```

Comment:
Instead of using this new global variable, please take a look to set/get_context methods